src/topology/generation.rs

```diff
@@ -0,0 +1,151 @@
+use crate::internal_events::InternalReloadFailed;
```

Comment:
Since you have already added doc comments to (almost) everything here, I recommend adding a `#![deny(missing_docs)]` to keep others on the right path. :grin: